### PR TITLE
feat: ✨ localization

### PR DESCRIPTION
**Pull request by Roaa94**
## Status

**IN DEVELOPMENT**

## Description

Adds localization by setting it up as instructed in the [official documentation](https://docs.flutter.dev/development/accessibility-and-localization/internationalization)

The only difference is that in `l10n.yaml` file, we had to set `synthetic-package` to false to generate local `app_localizations.dart` file that could be exported and used in other packages

```yaml
arb-dir: lib/src/l10n
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
synthetic-package: false
```

The setup is complete for `app_widgetbook` as well and languages can be switched between. 

An example to best preview the functionality widget would be `Core->Tweet->Detailed Tweet Metrics`

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: app_widgetbook/.gitignore [hunk 1]
```diff
@@ -42,3 +42,5 @@ app.*.map.json
 /android/app/debug
 /android/app/profile
 /android/app/release
+
+pubspec.lock
```
Comment (jenshor) [inline app_widgetbook/.gitignore]: This file should not be ignored. It's important.

File: app_widgetbook/pubspec.lock [hunk 2]
```diff
@@ -1,250 +0,0 @@
-# Generated by pub
-# See https://dart.dev/tools/pub/glossary#lockfile
-packages:
-  async:
-    dependency: transitive
-    description:
```
Comment (jenshor) [inline app_widgetbook/pubspec.lock]: This file should not be deleted.

File: packages/app/.gitignore [hunk 3]
```diff
@@ -42,3 +42,5 @@ app.*.map.json
 /android/app/debug
 /android/app/profile
 /android/app/release
+
+pubspec.lock
```
Comment (jenshor) [inline packages/app/.gitignore]: Is important.

File: packages/app/pubspec.lock [hunk 4]
```diff
@@ -1,160 +0,0 @@
-# Generated by pub
-# See https://dart.dev/tools/pub/glossary#lockfile
-packages:
-  async:
-    dependency: transitive
```
Comment (jenshor) [inline packages/app/pubspec.lock]: Should not be deleted.

**Review by jenshor — COMMENTED** [review]
The localization implementation looks fine to me.
However, we should make sure that the `pubspec.lock` file is included within `git` since it contains important information regarding package versions.

File: packages/app/pubspec.lock (hunk 4 above)
Comment (Roaa94) [reply packages/app/pubspec.lock]: Re-added

File: packages/app/.gitignore (hunk 3 above)
Comment (Roaa94) [reply packages/app/.gitignore]: Removed

File: app_widgetbook/pubspec.lock (hunk 2 above)
Comment (Roaa94) [reply app_widgetbook/pubspec.lock]: Re-added

File: app_widgetbook/.gitignore (hunk 1 above)
Comment (Roaa94) [reply app_widgetbook/.gitignore]: Removed

**Review by jenshor — APPROVED** [review]

**Pull request merged**